Fix openvpn3-linux builds

mattock opened the pull request:
This PR fixes openvpn3-linux builds insofar as it can be done with reasonable effort.

Several older distros were unable to (easily) compile gdbuspp (openvpn3-linux dependency) or had too old meson (buildsystem) version in the repositories. Fixing those older operating systems would have required lots of additional work, probably for little gain.

With the exception of Alpine Linux all the more recent systems compile just fine. There some library version may be wrong which prevents openvpn3-linux from compiling despite the setup phase succeeding. 

I bumped container image versions by one minor version up (e.g. 1.0.3 -> 1.1.0) for those distros where openvpn3-linux could be compiled. For the rest I added one to the patchlevel (e.g. 1.0.2 -> 1.0.3) because the images are no exactly the same as before due to extra (unused) dependencies such as "meson" having been installed on them.

I encountered an unrelated issue on Fedora Rawhide: when the container image is built from scratch, Buildmaster is unable to instantiate it because it hits this error when launching the Buildbot worker (i.e. twisted):

```
--- snip ---
  File "/buildbot_venv/lib/python3.9/site-packages/buildbot/process/buildstep.py", line 860, in runCommand
    res = yield command.run(self, self.remote, self.build.builder.name)
  File "/buildbot_venv/lib/python3.9/site-packages/buildbot/process/remotecommand.py", line 167, in _finished
    yield self.remoteComplete(failure)
buildbot.process.remotecommand.RemoteException: b'buildbot_worker.exceptions.AbandonChain': (-1, "Got exception (No module named 'pipes')")
Traceback (most recent call last):
Failure: buildbot_worker.exceptions.AbandonChain: (-1, "Got exception (No module named 'pipes')")
```

mattock committed:
Fix openvpn3-linux builds

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

flichtenheld commented:
Disabling openvpn3-linux for centos7 worker seems to be missing. Otherwise looks good to me.

dsommers commented on buildbot-host/scripts/install-openvpn-build-deps-fedora.sh:
```diff
@@ -10,6 +10,7 @@ automake \
 bzip2 \
 ccache \
 cmake \
+dbus-c++-devel \
```
This is not needed at all.  You will need `dbus-devel`, though.

dsommers commented on buildbot-host/scripts/install-openvpn-build-deps-ubuntu.sh:
```diff
@@ -35,8 +35,10 @@ libasio-dev \
 libcap-dev \
 libcap-ng-dev \
 libcmocka-dev \
+libdbus-c++-dev \
```
Same as above, replace with `libdbus-dev`

dsommers commented on buildbot-host/scripts/install-openvpn-build-deps-alpine.sh:
```diff
@@ -11,9 +11,11 @@ ccache \
 cmake \
 cmocka-dev \
 curl \
+dbus-cpp-dev \
```
This is not needed, but will need `dbus-dev` instead.

mattock replied:
Fixed. That said, Alpine builds continue to fail. I can report an issue about this if you want, but here's the build output from Buildbot:

```
[13/243] Compiling C++ object src/tests/getlaststatus.p/dbus_getlaststatus.cpp.o
ninja: job failed: ccache c++ -Isrc/tests/journal-log-parse.p -Isrc/tests -I../src/tests -I../openvpn3-core -I../vendor/asio/asio/include -Isrc -I../src -I. -I.. -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/gio-unix-2.0 -I/usr/include/blkid -I/usr/include/libmount -I/usr/include/libnl3 -I/usr/include/uuid -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++17 -O0 -g -Wno-non-virtual-dtor -DOVPN_TINYXML2_HAS_ERROR_NAME -DOVPN_TINYXML2_HAS_ERROR_STR -pthread -MD -MQ src/tests/journal-log-parse.p/misc_journal-log-parse-test.cpp.o -MF src/tests/journal-log-parse.p/misc_journal-log-parse-test.cpp.o.d -o src/tests/journal-log-parse.p/misc_journal-log-parse-test.cpp.o -c ../src/tests/misc/journal-log-parse-test.cpp
../src/tests/misc/journal-log-parse-test.cpp:19:17: error: 'Log' has not been declared
   19 | using namespace Log::Journald;
      |                 ^~~
../src/tests/misc/journal-log-parse-test.cpp:19:22: error: 'Journald' is not a namespace-name
   19 | using namespace Log::Journald;
      |                      ^~~~~~~~
../src/tests/misc/journal-log-parse-test.cpp: In function 'int main(int, char**)':
../src/tests/misc/journal-log-parse-test.cpp:23:5: error: 'Parse' was not declared in this scope
   23 |     Parse jrn;
      |     ^~~~~
../src/tests/misc/journal-log-parse-test.cpp:43:9: error: 'Parse' is not a class, namespace, or enumeration
   43 |         Parse::FilterType ft = {};
      |         ^~~~~
../src/tests/misc/journal-log-parse-test.cpp:46:13: error: 'ft' was not declared in this scope
   46 |             ft = Parse::FilterType::TIMESTAMP;
      |             ^~
../src/tests/misc/journal-log-parse-test.cpp:46:18: error: 'Parse' is not a class, namespace, or enumeration
   46 |             ft = Parse::FilterType::TIMESTAMP;
      |                  ^~~~~
../src/tests/misc/journal-log-parse-test.cpp:50:13: error: 'ft' was not declared in this scope
   50 |             ft = Parse::FilterType::LOGTAG;
      |             ^~
../src/tests/misc/journal-log-parse-test.cpp:50:18: error: 'Parse' is not a class, namespace, or enumeration
   50 |             ft = Parse::FilterType::LOGTAG;
      |                  ^~~~~
../src/tests/misc/journal-log-parse-test.cpp:54:13: error: 'ft' was not declared in this scope
   54 |             ft = Parse::FilterType::SESSION_TOKEN;
      |             ^~
../src/tests/misc/journal-log-parse-test.cpp:54:18: error: 'Parse' is not a class, namespace, or enumeration
   54 |             ft = Parse::FilterType::SESSION_TOKEN;
      |                  ^~~~~
../src/tests/misc/journal-log-parse-test.cpp:58:13: error: 'ft' was not declared in this scope
   58 |             ft = Parse::FilterType::OBJECT_PATH;
      |             ^~
../src/tests/misc/journal-log-parse-test.cpp:58:18: error: 'Parse' is not a class, namespace, or enumeration
   58 |             ft = Parse::FilterType::OBJECT_PATH;
      |                  ^~~~~
../src/tests/misc/journal-log-parse-test.cpp:62:13: error: 'ft' was not declared in this scope
   62 |             ft = Parse::FilterType::SENDER;
      |             ^~
../src/tests/misc/journal-log-parse-test.cpp:62:18: error: 'Parse' is not a class, namespace, or enumeration
   62 |             ft = Parse::FilterType::SENDER;
      |                  ^~~~~
../src/tests/misc/journal-log-parse-test.cpp:66:13: error: 'ft' was not declared in this scope
   66 |             ft = Parse::FilterType::INTERFACE;
      |             ^~
../src/tests/misc/journal-log-parse-test.cpp:66:18: error: 'Parse' is not a class, namespace, or enumeration
   66 |             ft = Parse::FilterType::INTERFACE;
      |                  ^~~~~
../src/tests/misc/journal-log-parse-test.cpp:73:9: error: 'jrn' was not declared in this scope
   73 |         jrn.AddFilter(ft, value);
      |         ^~~
../src/tests/misc/journal-log-parse-test.cpp:73:23: error: 'ft' was not declared in this scope
   73 |         jrn.AddFilter(ft, value);
      |                       ^~
../src/tests/misc/journal-log-parse-test.cpp:76:5: error: 'LogEntries' was not declared in this scope
   76 |     LogEntries l = jrn.Retrieve();
      |     ^~~~~~~~~~
../src/tests/misc/journal-log-parse-test.cpp:78:18: error: 'l' was not declared in this scope
   78 |     std::cout << l.GetJSON() << std::endl;
      |                  ^
ninja: subcommand failed
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /buildbot/alpine-openvpn3-linux-openssl/build/_builddir
program finished with exit code 1
elapsedTime=157.954097
```

mattock replied:
I'll resolve this comment as the build failure is unrelated to the actual package name.

dsommers reviewed (changes requested):
The `dbus-c++` library does some of the same things as GDBus++, but with a bit different approach. And for many years it was basically dead, but seems to have been resurrected a couple of times.

But the OpenVPN 3 Linux needs `dbus-devel`, to get the `dbus-1.pc` file which contains all the various install paths to use for the support files.

mattock committed:
Disable openvpn3-linux builds on CentOS 7

They have zero chance of working

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock committed:
Fix dbus dependency packages for openvpn3-linux

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock committed:
Do not fail if pip cannot be upgraded in install-buildbot.sh

This can happen on some operating systems like Ubuntu 24.04 when pip was
installed from Debian packages.

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock commented:
> Disabling openvpn3-linux for centos7 worker seems to be missing. Otherwise looks good to me.

Fixed in a separate commit.
